README.md

```diff
@@ -155,21 +155,41 @@ and in your `configuration.nix`
 {inputs, pkgs, ... }:
 let
   inherit (pkgs.stdenv.hostPlatform) system;
-  umu = inputs.umu.packages.${system}.umu.override {
-    version = inputs.umu.shortRev;
-    truststore = true;
-    cbor2 = true;
-  };
+  unu-launcher = inputs.umu.packages.${system}.default;
 in
 {
-  environment.systemPackages = [ umu ];
+  environment.systemPackages = [ umu-launcher ];
 }
 ```
-> [!NOTE]
-> truststore and cbor2 (for delta updates) are optional dependency which are enabled by default if you want to keep it that way you can remove the `truststore = true; cbor2 = true;` part
+
+> [!TIP]
+> It is common practice to override umu's `version` to `inputs.umu.shortRev` (the umu git commit).
+>
+> ```nix
+>  umu-launcher = inputs.umu.packages.${system}.default.override {
+>    version = inputs.umu.shortRev;
+>  };
+> ```
+>
+> By default, the version is based on `inputs.umu`'s last modified date, e.g. `"git.2025.01.22.09.59.33"`.
+
+> [!TIP]
+> You can override `withTruststore` and/or `withDeltaUpdates` to disable optional dependencies.
+>
+> ```nix
+>  umu-launcher = inputs.umu.packages.${system}.default.override {
+>    withTruststore = false;
+>    withDeltaUpdates = false;
+>  };
+> ```
+>
+> - `withTruststore` adds dependencies that allow using the system trust store
+> - `withDeltaUpdates` adds dependencies that enable "delta updates" to Proton builds
+>
+> Both options are true by default.
```

Comment:
I don't exactly see why this is recommneded, but nor do I see why it's not the default. Why don't we just follow the nixpkgs pattern of doing `0-unstable-<pname>-<commit hash>` instead of the complicated pattern matching?

Reply:
You've selected such a large range of code, it's hard to tell whether you're just commenting on the version or also have issues with the `withSomething` args?

Regarding the version, I agree: using the revision hash is better. Good spot!

Reply:
I was referring to overriding the version - no idea why anyone would want to do that.

Reply:
Historically, this flake didn't set a useful version. Instead the README recommended users override it themselves. Seeing as we can set it to `sourceInfo.shortRev`, that is no longer necessary, hence why I've now removed that section of the README.

---

I used `lastModifiedDate` originally because `shortRev` isn't always set. With my initial testing I loaded the flake impurely via `builtins.getFlake`, so `sourceInfo` only had:

``{ lastModified, lastModifiedDate, narHash, outPath }``

However that's a very niche scenario... I'll probably still add an extra `or` fallback to be safe though.